# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasession/MediaSessionIPCUtils.h
# Commit: a008d3cebfdb
# Full Hash: a008d3cebfdb9d96ab32530b90180141575316a8
# Author: alwu <alwu@mozilla.com>
# Date: 2020-03-17 09:36:40
# Regressor Bug: 1582508
# File Overlap Count: 1
# Description:
#   Bug 1582508 - part4 : propagate declared playback state to media session controller. r=chunmin
#   
#   This patch includes the implementation of propagating declared playback state from the media session in the content process to the media session controller in the chrome process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D66341
# ==============================================================================

diff -r a73415e9c006 -r a008d3cebfdb dom/media/mediasession/MediaSessionIPCUtils.h
--- a/dom/media/mediasession/MediaSessionIPCUtils.h	Tue Mar 17 00:03:44 2020 +0000
+++ b/dom/media/mediasession/MediaSessionIPCUtils.h	Tue Mar 17 00:17:36 2020 +0000
@@ -64,6 +64,13 @@
   }
 };
 
+template <>
+struct ParamTraits<mozilla::dom::MediaSessionPlaybackState>
+    : public ContiguousEnumSerializer<
+          mozilla::dom::MediaSessionPlaybackState,
+          mozilla::dom::MediaSessionPlaybackState::None,
+          mozilla::dom::MediaSessionPlaybackState::EndGuard_> {};
+
 }  // namespace IPC
 
 #endif  // DOM_MEDIA_MEDIASESSION_MEDIASESSIONIPCUTILS_H_